docs/developer-guide/udf.rst

```diff
@@ -233,20 +234,30 @@ of the UDF does, for example:
 UdfParameter Annotation
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``@UdfParameter`` annotation is optional and is applied to the parameters of methods annotated with
-``@Udf``. KSQL will use the additional information in the ``@UdfParameter`` annotation to provide
-users with richer information about the method when, for example, they execute
-``DESCRIBE FUNCTION`` on the method.
-
-The annotation has two parameters: ``value`` is the name of the parameter and ``description`` which
-can be used to better describe what the parameter does, for example:
+The ``@UdfParameter`` annotation is applied to parameters of methods annotated with ``@Udf``. KSQL
+will use the additional information in the ``@UdfParameter`` annotation to specify the parameter
+schema (if it cannot be inferred from the Java type) or to provide users with richer information
+about the method when, for example, they execute ``DESCRIBE FUNCTION`` on the method.
+
++------------+------------------------------+------------------------+
+| Field      | Description                  | Required               |
++============+==============================+========================+
+| value      | The case-insensitive name of | No                     |
```

Comment:
Thinking on this more - I actually think it should be an error to have unnamed parameters.  How is the user supposed to know what they are?  I think KSQL should throw an exception if `value` is not supplied and the jar isn't compiled with `-parameters`.  (Outside of scope for this PR...)

Reply:
I can address this in a follow-up PR since I don't feel that strongly about it (though that would break compile-time backwards compatibility) - though if the user doesn't care, why force them to?

Reply:
It's about the help that is displayed to the user of the UDF.  The UDF dev may be lazy and just not provide param names and not compile with `-parameters`... which you could argue is their choice.  But then the user experience in KSQL isn't great.

IMHO, better to require one or the other. 

But, yes, a different PR,

Reply:
https://github.com/confluentinc/ksql/pull/2863